bindings/cs/MongoDB.Crypt/MongoCryptSafeHandle.cs

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019–present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+#if NET452 || NETSTANDARD2_0
+using System.Runtime.ConstrainedExecution;
+#endif
+using System.Runtime.InteropServices;
+
+namespace MongoDB.Crypt
+{
+    /// <summary>
+    /// SafeHandle to manage the lifetime of a mongocrypt_t.
+    /// </summary>
+    /// <seealso cref="System.Runtime.InteropServices.SafeHandle" />
+    internal class MongoCryptSafeHandle : SafeHandle
+    {
+        private MongoCryptSafeHandle()
+            : base(IntPtr.Zero, true)
+        {
+        }
```

Comment:
Where does the handle ever get set?

Reply:
It is set by calling the delegate mongocrypt_new I believe (sorry not at my desk to look it up)

Reply:
Still not following.

Doesn't the handle need to be passed in to the constructor?

Reply:
Even though I don't understand the nuances of how SafeHandle works since Mark says the handle get sets somewhere else (and not the constructor) I'm resolving this comment.